From c52751a2d825314adb1178e007e1337f533bf621 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Mon, 2 Mar 2015 14:09:29 +0300
Subject: [PATCH 11/N] convert(): use the elegant realloc() call

Instead of doing a poor man's realloc(), we can call the real thing ;-)

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/path.cc | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/winsup/cygwin/path.cc b/winsup/cygwin/path.cc
index 22135c5..2ded99d 100644
--- a/winsup/cygwin/path.cc
+++ b/winsup/cygwin/path.cc
@@ -3525,8 +3525,7 @@ arg_heuristic_with_exclusions (char const * const arg, char const * exclusions,
     {
       return ((char *)arg);
     }
-  arg_result = (char *)malloc (strlen (stack_path)+1);
-  strcpy (arg_result, stack_path);
+  arg_result = (char *)realloc (stack_path, strlen (stack_path)+1);
   // Windows doesn't like empty entries in PATH env. variables (;;)
   char* semisemi = strstr(arg_result, ";;");
   while (semisemi)
-- 
2.9.0

